src/fromager/settings.py

```diff
@@ -82,6 +82,12 @@ def resolver_include_sdists(
             resolve_dist.get("include_sdists"), default
         )
 
+    def build_dir(self, pkg: str, sdist_root_dir: pathlib.Path) -> pathlib.Path:
+        p = self.get_package_settings(pkg)
+        if p.get("build_dir"):
+            return sdist_root_dir / p.get("build_dir")
```

Comment:
We should do something here to ensure that this relative path does not go outside of the sdist_root_dir. For example, if sdist_root_dir is `/foo/bar` and someone sets the build_dir to `/tmp/foo` we need to ensure that resolves to `/foo/bar/tmp/foo` and not `/tmp/foo`.

```
>>> import pathlib
>>> p = pathlib.Path('/foo/bar')
>>> t = pathlib.Path('/tmp/blah')
>>> p / t
PosixPath('/tmp/blah')
```